test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Actions/UIActionEngineTests.cs

```diff
@@ -162,39 +162,39 @@ public async Task GetPreviewResultsAsync_WithMultipleActions_SortsPackageIdentit
             Assert.Equal(packageIdentityC.Id, addedResults[2].Id);
         }
 
-        private UIActionEngine CreateUIActionEngine(ISettings settings = null)
+        private UIActionEngine CreateUIActionEngine(ISettings settings = null, INuGetTelemetryProvider nuGetTelemetryProvider = null)
         {
             settings = settings ?? Mock.Of<ISettings>();
+            nuGetTelemetryProvider = nuGetTelemetryProvider ?? Mock.Of<INuGetTelemetryProvider>();
             var sourceRepositoryProvider = Mock.Of<ISourceRepositoryProvider>();
             var packageManager = new NuGetPackageManager(
                 sourceRepositoryProvider,
-                Mock.Of<ISettings>(),
+                settings,
                 TestDirectory.Create().Path);
 
             return new UIActionEngine(
                 sourceRepositoryProvider,
                 packageManager,
-                Mock.Of<INuGetLockService>());
+                Mock.Of<INuGetLockService>(),
+                nuGetTelemetryProvider);
         }
 
         [Theory]
         [MemberData(nameof(GetInstallActionTestData))]
         public async Task CreateInstallAction_OnInstallingProject_WithNewSourceMapping_DoesNotLogTelemetry(ContractsItemFilter activeTab, bool isSolutionLevel, string packageIdToInstall, bool? expectedPackageWasTransitive)
         {
             // Arrange
-            var telemetrySession = new Mock<ITelemetrySession>();
             TelemetryEvent lastTelemetryEvent = null;
-            telemetrySession
-                .Setup(x => x.PostEvent(It.IsAny<TelemetryEvent>()))
+            var telemetryProvider = new Mock<INuGetTelemetryProvider>();
+            telemetryProvider
+                .Setup(x => x.EmitEvent(It.IsAny<TelemetryEvent>()))
```

Comment:
I still don't like the setup we had here where the _last_ event has to be XYZ. Why would we require it to be the most recent event? Regardless, if this change fixes that, that's the main goal!

Reply:
Yeah, it's the only event we use, so I guess it's ok. 
Not in love with it, but not point is storing more events. 

> Do you know if the event status is now something besides Cancelled after this change? I had to remove a bunch of assertions due to that in https://github.com/NuGet/Client.Engineering/issues/2798.

Still Cancelled. 

Root cause is https://github.com/NuGet/NuGet.Client/blob/843d21ef3560d0f7655a5ddadf0987b8e50a2041/src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs#L368-L372, the service is unavailable there and this is what it returns.